BraveWallet/Crypto/Stores/KeyringStore.swift

```diff
@@ -367,9 +371,13 @@ extension KeyringStore: BraveWalletKeyringServiceObserver {
 
   public func selectedAccountChanged(_ coinType: BraveWallet.CoinType) {
     Task { @MainActor in
+      let previouslySelectedCoin = await walletService.selectedCoin()
       walletService.setSelectedCoin(coinType)
-      let network = await rpcService.network(coinType)
-      await rpcService.setNetwork(network.chainId, coin: coinType) // update network here in case NetworkStore is closed. p.s. Multiple network selection with the same network will not cause `chainChangedEvent` getting called multiple times.
+      if previouslySelectedCoin != coinType {
+        // update network here in case NetworkStore is closed.
+        let network = await rpcService.network(coinType)
+        await rpcService.setNetwork(network.chainId, coin: coinType)
+      }
```

Comment:
dont think networkStore will ever close while keyringStore is open and getting `selectedAccountChanged` called.

Reply:
I think `KeyringStore` can be initialized without `NetworkStore` in `WalletStore` if the default keyring is not created. This was added when we [added Solana account creation](https://github.com/brave/brave-ios/pull/5705/files#diff-adaa11a29db87248efa6499c3d9f1c020b1461ee06dfcf56b3f31bea2c6ee0c2R378).